### PR TITLE
added data_format argument to neccessary functions within layers.py

### PR DESCRIPTION
**Pull request by mosesdaudu001**
# PR Description

### CODE REVIEW

**Comment by mosesdaudu001** [comment]
This task is to add the data_format argument to neccessary functions within the layers.py file.

[https://trello.com/c/Quq386m4/2422-add-dataformat-argument-to-whichever-function-from-the-layers-submodule-is-missing](url)

**Comment by ivy-leaves** [comment]
If you are working on an open task, please **edit** the PR description to link to the issue you've created.

For more information, please check [ToDo List Issues Guide](https://unify.ai/docs/ivy/overview/contributing/the_basics.html#todo-list-issues).

Thank you :hugs:

**Comment by ivy-leaves** [comment]
If you are working on an open task, please **edit** the PR description to link to the issue you've created.

For more information, please check [ToDo List Issues Guide](https://unify.ai/docs/ivy/overview/contributing/the_basics.html#todo-list-issues).

Thank you :hugs:

**Comment by AnnaTz** [comment]
Hey @mosesdaudu001, you will also need to update the testing of the modified functions so that we can check they are behaving correctly.

**Comment by mosesdaudu001** [comment]
> Hey @mosesdaudu001, you will also need to update the testing of the modified functions so that we can check they are behaving correctly.

For some reason I can't seem to test any function within the layers submodule of the experimental API, as it doesn't appear within the tests. I tried both locally and on codespaces. Are you also facing the same issue?
Screenshot (130)

**Comment by mosesdaudu001** [comment]
> Hey @mosesdaudu001, you will also need to update the testing of the modified functions so that we can check they are behaving correctly.

For some reason I can't seem to test any function within the layers submodule of the experimental API, as it doesn't appear within the tests. I tried both locally and on codespaces. Are you also facing the same issue?
Screenshot (130)

**Comment by AnnaTz** [comment]
> > Hey @mosesdaudu001, you will also need to update the testing of the modified functions so that we can check they are behaving correctly.
> 
> For some reason I can't seem to test any function within the layers submodule of the experimental API, as it doesn't appear within the tests. I tried both locally and on codespaces. Are you also facing the same issue? Screenshot (130)

I can run the tests without any issues in pycharm. Maybe you should ask about this on discord.

**Review by vedpatwardhan — COMMENTED** [review]
I'll leave this with @sherry30 and @AnnaTz to review this as it's related to layers. Thanks @mosesdaudu001 😄

File: ivy/functional/backends/torch/experimental/layers.py [hunk 1]
```diff
@@ -927,7 +927,11 @@ def adaptive_avg_pool1d(input, output_size):
 
 
 @with_unsupported_dtypes({"2.0.1 and below": ("bfloat16", "float16")}, backend_version)
-def adaptive_avg_pool2d(input, output_size):
+def adaptive_avg_pool2d(
+    input,
+    output_size,
+    data_format: str = "NCHW",
+):
     return torch.nn.functional.adaptive_avg_pool2d(input, output_size)
```
Comment (AnnaTz) [inline ivy/functional/backends/torch/experimental/layers.py:935]: You don't handle `data_format` at all here. And you also haven't added it to `adaptive_avg_pool1d`. Please complete the code and make sure to run the tests locally to confirm the changes work.

@vedpatwardhan there must be a problem in the CI checks since the test is shown to be passing.

Comment (vedpatwardhan) [reply ivy/functional/backends/torch/experimental/layers.py:935]: Yeah there have been a few issues with the intelligent tests, but the `adaptive_avg_pool2d` test seems to have [run](https://github.com/unifyai/ivy/actions/runs/6160405092/job/16717945740?pr=23065#step:3:8017), so the problem isn't with the discovery of tests which is the main problem in most other PRs. I'm not sure why the test is passing though, does it fail when you run the test with `25` examples (as that's the max_examples used in the intelligent tests)? Thanks @AnnaTz 😄

Comment (AnnaTz) [reply ivy/functional/backends/torch/experimental/layers.py:935]: Yes, it fails locally.

**Comment by ivy-seed** [comment]
This PR has been labelled as stale because it has been inactive for more than 7 days. If you would like to continue working on this PR, then please add another comment or this PR will be closed in 7 days.

**Comment by ivy-seed** [comment]
This PR has been closed because it has been marked as stale for more than 7 days with no activity.
